### DISCUSSION TITLE
How to mask (hide) details and trees inside a volume

### DISCUSSION THREAD

**Discussion by Marshal974** [visualdesigncafe/nature-renderer#90] [Help]
Hi, I used to use VegetationStudio for that but it is obsolete. 
It had a component called "vegetation mask" that was hiding all nature (details and trees) inside a collider. 
I am looking for an alternative to that. 

I tried creating scripts that works when it comes to cleaning the terrain, the trees can be displayed back when quitting the volume, but it does not work for the terrain details. 

What is the best approach to hide temporarily at run time some vegetation elements using NatureRenderer ? 

Thank you very much !

**Comment by maxplooi** [comment]
There is no direct support for this, but you can achieve the effect with some custom code in one of the following ways:
1. Edit the terrain data at runtime to (temporarily) remove the details. You can do this using the regular Unity Terrain API. There is an example in the docs: https://visualdesigncafe.com/nature-renderer/docs/2022/terrain-vegetation#scripting-api
2. Use a custom shader that hides the vegetation when inside of a mask. Our [Nature Shaders](https://visualdesigncafe.com/nature-shaders/) asset has support for this but it does not yet support Unity 6.
